services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java

```diff
@@ -67,6 +67,18 @@ public int deleteRuleByTitle(String ruleTitle) throws PersistenceException {
 
     }
 
+
+    public List<String>  getDomainList() throws PersistenceException {
+        TypedQuery<String> query = getEntityManager().createNamedQuery("Rule.getDomainList", String.class);
+
+        try {
+            return query.getResultList();
+        } catch (NoResultException e) {
+            return null;
```

Comment:
No log needed here?

Reply:
Maybe a TU should be updated to cover this new method

Reply:
- No log needed here we just need to catch the  NoResultException so the method will return null when the list is empty

- Developing the Unit tests is in progress in another user story, this can be added as requirement threr